### PR TITLE
Remove methods that detected wrong classes loaded

### PR DESCRIPTION
**Pull request by nicolasstucki**
Now we load them from their .tasty which ensures that the class was generated by Scala 3 (not Scala 2 nor Java).

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/fromtasty/ReadTasty.scala [hunk 1]
```diff
@@ -44,9 +44,8 @@ class ReadTasty extends Phase {
                 Some(unit)
               }
             case tree: Tree[?] =>
+              // TODO handle correctly this case correctly to get the tree or avoid it completely.
               cls.denot.infoOrCompleter match {
-                case _: NoLoader => Some(Scala2CompilationUnit(cls.fullName.toString))
-                case _ if cls.flags.is(Flags.JavaDefined) => Some(JavaCompilationUnit(cls.fullName.toString))
                 case _ => Some(AlreadyLoadedCompilationUnit(cls.denot.fullName.toString))
               }
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/fromtasty/ReadTasty.scala:50]: The `match` is kept intentionally? Semantically, it can be removed.

**Review by liufengyun — APPROVED** [review]
LGTM

File: compiler/src/dotty/tools/dotc/fromtasty/ReadTasty.scala (hunk 1 above)
Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/fromtasty/ReadTasty.scala:50]: We should be able to remove it. But last time I tried there was a failure with some of the `-from-tasty` tests. As this is unrelated to the public API I left it for a second PR.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/fromtasty/ReadTasty.scala:50]: I also left the comment above to not forget to remove this

**Pull request merged**